lib/imports/api/authority-registry/server/authority-registry.js

```diff
@@ -0,0 +1,319 @@
+  * @returns {{}} authority
+  */
+  getAuthorityInfoByDocId ({ docId }) {
+    return authorityRegistryColl.findOne({ docIds: docId });
+  }
+
+  /**
+  *  Returns a single authority from the registry database, found via a groupId that lives in an authority.
+  * @param {Object} params
+  * @param {String} params.groupId        groupId that the wanted authority is hosting
+  * @returns {{}} authority
+  */
+  getAuthorityInfoByGroupId ({ groupId }) {
+    return authorityRegistryColl.findOne({ groupIds: groupId });
+  }
+  /**
+  *  Returns if the specified server is registered or not.
+  * @param {Object} params
+  * @param {String} params.ip     ip of the server to remove from the registry
+  * @param {Number} params.port   port of the server to remove from the registry
+  * @returns {Boolean} if the server is registered
+  */
+  isRegistered ({ ip, port }) {
+    check(ip, String);
+    check(port, Number);
+    return authorityRegistryColl.find({ ip, port }).count() !== 0;
+  }
+
+  /**
+  *  Adds a doc to an authority in the registry.
+  * @param {Object} params
+  * @param {String} params.docId      unique id for this doc
+  * @param {String} [params.groupId]  optional, a groupId of this doc to associate it with other docs
+  */
+  addDocToAuthority ({ docId, groupId = null }) {
+    addDocToAuthority.call({ docId, groupId, fullUrl: this.fullUrl });
+    console.log(`ProseMeteor AuthorityRegistry: Added doc ${docId} to authority ${this.fullUrl}. Updated registry: ${JSON.stringify(this.fullRegistry(), null, 2)}`);
+  }
+  /**
+  *  Removes a doc from an authority in the registry. If this doc is in a groupId and no more of that group are hosted
+  *  on this server, the groupId is also removed from this authority.
+  * @param {Object} params
+  * @param {String} params.docId      unique id for this doc
+  * @param {String} [params.groupId]  optional, a groupId of this doc to associate it with other docs
+  */
+  removeDocFromAuthority ({ docId, groupId = null }) {
+    // first remove this docId
+
+    if (groupId) {
+      // check each remaining docId. if any have the same groupId, keep this docs groupId. otherwise remove it from groupIds[]
+    }
+  }
+  /**
+  *  Chooses a list of eligible authority servers to host a document, preferring the server with the least documents.
+  * @returns {Array}  chosenAuthorities     array of chosen authority servers
+  */
+  getEligibleAuthoritiesToHostDoc () {
+    // otherwise get 1-5 servers hosting the lowest number of docs
+    const fullRegistry = this.fullRegistry();
+    const fullRegistrySortedByDocIdsCount = fullRegistry
+      .sort((authorityA, authorityB) => {
+        return authorityA.docIds.length > authorityB.docIds.lenth;
+      })
+      .slice(0, 5);
+    console.log(`ProseMeteor AuthorityRegistry: Selecting eligible authority servers with least documents: "${JSON.stringify(fullRegistrySortedByDocIdsCount, null, 2)}"`);
+
```

Comment:
This is really hard to answer without testing it on a few servers under load, and even harder because it would depend on the application user's implementation. Right now all of the db load is when a user connects to a document to query the db to see if any other servers are hosting the doc, or to delegate a document to a server. Once users are connected to a server, it's all just reading and writing from memory for every keystroke so that won't be much db load and JavaScript is pretty damn fast for stuff like that. 

I think at the end of the day, we can't predict how this package will be used and what the right scaling decisions will be that work for everyone, so our best choice is to allow the application developer to specify their own options for things like max number of users who can edit a doc, snapshot interval time, etc, with some intelligent defaults that we provide that would work for basic usage.